app/src/main/java/org/openobservatory/ooniprobe/activity/OverviewActivity.java

```diff
@@ -179,6 +194,72 @@ private void setUpOnCLickListeners() {
+                Data.Builder data = new Data.Builder();
+                data.putLongArray(ManualUpdateDescriptorsWorker.KEY_DESCRIPTOR_IDS, new long[]{Objects.requireNonNull(descriptor.getDescriptor()).getRunId()});
+                OneTimeWorkRequest manualWorkRequest = new OneTimeWorkRequest.Builder(ManualUpdateDescriptorsWorker.class)
+                        .setConstraints(
+                                new Constraints.Builder()
+                                        .setRequiredNetworkType(NetworkType.CONNECTED)
+                                        .build()
+                        ).setInputData(data.build())
+                        .build();
+
+                WorkManager.getInstance(this)
+                        .beginUniqueWork(
+                                ManualUpdateDescriptorsWorker.UPDATED_DESCRIPTORS_WORK_NAME,
+                                ExistingWorkPolicy.REPLACE,
+                                manualWorkRequest
+                        ).enqueue();
+
+                WorkManager.getInstance(this)
+                        .getWorkInfoByIdLiveData(manualWorkRequest.getId())
+                        .observe(this, this::onManualUpdatesFetchComplete);
+
+            });
+        } else {
+            binding.swipeRefresh.setEnabled(false);
+        }
+    }
+
+
+    /**
+     * Listens to updates from the {@link ManualUpdateDescriptorsWorker}.
+     * <p>
+     * This method is called after the {@link ManualUpdateDescriptorsWorker} is enqueued.
+     * The {@link ManualUpdateDescriptorsWorker} task is to fetch updates for the descriptors.
+     * <p>
+     * If the task is successful, the {@link WorkInfo} object will contain the updated descriptors.
+     * Otherwise, the {@link WorkInfo} object will be null.
+     *
+     * @param workInfo The {@link WorkInfo} of the task.
+     */
+    private void onManualUpdatesFetchComplete(WorkInfo workInfo) {
+        if (workInfo != null) {
+            switch (workInfo.getState()) {
+                case SUCCEEDED -> {
+                    binding.reviewUpdates.setVisibility(View.VISIBLE);
+                    binding.reviewUpdates.setOnClickListener(view -> getReviewUpdatesLauncher().launch(
+                            ReviewDescriptorUpdatesActivity.newIntent(
+                                    OverviewActivity.this,
+                                    workInfo.getOutputData().getString(ManualUpdateDescriptorsWorker.KEY_UPDATED_DESCRIPTORS)
+                            )
+                    ));
+                    binding.swipeRefresh.setRefreshing(false);
+                }
+                case FAILED -> {
+                    binding.swipeRefresh.setRefreshing(false);
+                    Snackbar.make(
+                            binding.getRoot(),
+                            R.string.Modal_Error,
+                            Snackbar.LENGTH_LONG
+                    ).show();
+                }
+                default -> {
```

Comment:
```suggestion

                default -> {
```

As commented above, we should explain what is the default case--no updates?